src/it/scala/net/snowflake/spark/snowflake/SimpleNewPushdownIntegrationSuite.scala

```diff
@@ -0,0 +1,227 @@
+                     |	(SELECT * FROM $test_table2
+                     |) AS "subquery_0"
+                     | WHERE ((("subquery_0"."P" IS NOT NULL) AND ("subquery_0"."P" > 1)) AND ("subquery_0"."P" < 3))
+      """.stripMargin,
+                 result,
+                 Seq(Row(2, 2), Row(3, 2)))
+  }
+
+  test("LIMIT and SORT") {
+    val result =
+      sparkSession.sql("""
+        select * from df2
+        where p > 1 AND p < 3 order by p desc limit 1
+      """.stripMargin)
+
+    testPushdown(s"""SELECT * FROM
+                     |	(SELECT * FROM
+                     |		(SELECT * FROM
+                     |			(SELECT * FROM $test_table2
+                     |		) AS "subquery_0"
+                     |	 WHERE ((("subquery_0"."P" IS NOT NULL) AND ("subquery_0"."P" > 1)) AND ("subquery_0"."P" < 3))
+                     |	) AS "subquery_1"
+                     | ORDER BY ("subquery_1"."P") DESC
+                     |) AS "subquery_2"
+                     | LIMIT 1
+      """.stripMargin,
+                 result,
+                 Seq(Row(2, 2)))
+  }
+
+  test("Nested query with column alias") {
+    val result =
+      sparkSession.sql("""
+        select f, o from (select o, p as f from df2
+        where p > 1 AND p < 3) as foo order by f,o desc
+      """.stripMargin)
+
+    testPushdown(
+      s"""SELECT * FROM
+          |	(SELECT ("subquery_1"."P") AS "f", "subquery_1"."O" FROM
+          |		(SELECT * FROM
+          |			(SELECT * FROM $test_table2
+          |		) AS "subquery_0"
+          |	 WHERE ((("subquery_0"."P" IS NOT NULL) AND ("subquery_0"."P" > 1)) AND ("subquery_0"."P" < 3))
+          |	) AS "subquery_1"
+          |) AS "subquery_2"
+          | ORDER BY ("subquery_2"."f") ASC, ("subquery_2"."O") DESC
+      """.stripMargin,
+      result,
+      Seq(Row(2, 3), Row(2, 2)))
+  }
+
+  test("Sum and RPAD") {
+    val result =
+      sparkSession.sql(
+        """
+        select sum(i) as hi, rpad(s,10,"*") as ho from df1 group by s
+                       """.stripMargin)
+
+    testPushdown(
+      s"""SELECT (SUM("subquery_0"."I")) AS "hi", (RPAD("subquery_0"."S", 10, '*')) AS "ho" FROM
+          |		(SELECT * FROM $test_table
+          |	) AS "subquery_0"
+          | GROUP BY "subquery_0"."S"
+      """.stripMargin,
+      result,
+      Seq(Row(null, "Hello*****"),
+          Row(2, "Snowflake*"),
+          Row(3, "Spark*****"),
+          Row(4, null)))
+
+  }
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+  }
+
+  override def afterAll(): Unit = {
+    try {
+      jdbcUpdate(s"drop table if exists $test_table")
+      jdbcUpdate(s"drop table if exists $test_table2")
+    } finally {
+      super.afterAll()
+      SnowflakeConnectorUtils.disablePushdownSession(sqlContext.sparkSession)
+    }
+  }
+
+  /**
+    * Verify that the pushdown was done by looking at the generated SQL,
+    * and check the results are as expected
+    */
+  def testPushdown(reference: String,
```

Comment:
You have another, almost identical testPushdown() in another file, maybe merge them